web/src/stream_popover.js

```diff
@@ -577,6 +577,18 @@ export function build_move_topic_to_stream_popover(current_stream_id, topic_name
         $("#move_topic_modal .inline_topic_edit").on("input", () => {
             update_submit_button_disabled_state(stream_widget.value());
         });
+
+        setTimeout(() => {
+            if (!disable_stream_input && args.disable_topic_input) {
+                $("#select_stream_widget .button").trigger("focus");
+            } else {
+                $(".inline_topic_edit").trigger("focus");
+                $(".inline_topic_edit")[0].setSelectionRange(
```

Comment:
Maybe in future we can add a helper function `focus_on_text_input` also.

Reply:
>One other way to do this can be to save the input text in a variable first and then clear the input and then set the text for the input again.

I knew of another way to do this, which involves saving the input text in a variable, clearing the input, and then setting the text for the input again. However, I found using setSelectionRange to be a better way because it only requires one line of code whereas other require 3 lines and an extra variable. 
Both methods set the cursor at the end of the input for long text, but it may not be visible. I tried to find a way to shift the text and make the cursor visible, but was unsuccessful. If you know how to achieve this, please feel free to share. 

Also, I added the reason for my change in the commit message, but I will add a comment to explain it further.

Reply:
You might be able to just use `$.caret(-1)` to set the cursor position? Not sure if that'll be off by one.

Reply:
> Maybe in future we can add a helper function `focus_on_text_input` also.

@sahil839 in which file I should create this such that it can easily be globally used in any other file?
and is the name `focus_on_text_input` is final for the function or is it just a placeholder for now?

Reply:
> Maybe in future we can add a helper function `focus_on_text_input` also.

@sahil839, created this function in PR #24958, I will update this PR to use that function once it is merged.